src/components/common/DragAndDrop/DragAndDrop.jsx

```diff
@@ -33,30 +33,31 @@ const DragAndDrop = ({onFilesSelected, updateUploadedFiles}) => {
     const droppedFiles = e.dataTransfer.files;
     if (droppedFiles && droppedFiles[0]) {
       const newFiles = Array.from(droppedFiles);
-      setFiles((prevFiles) => [...prevFiles, ...newFiles]);
-      console.log("DROPPED", files);
-      memoizedUpdateUploadedFiles(newFiles);
+      // setFiles((prevFiles) => [...prevFiles, ...newFiles]);
+      // console.log("DROPPED", files); // Note: don't log here (logs previous state, so will not be accurate)
+      // memoizedUpdateUploadedFiles(newFiles);
+      updateUploadedFiles(prev => [...prev, ...newFiles]);
     }
   };
   
-  const handleChange = function (e) {
-    e.preventDefault();
-    const selectedFiles = e.target.files;
-    if (selectedFiles && selectedFiles[0]) {
-      const newFiles = Array.from(selectedFiles);
-      setFiles((prevFiles) => [...prevFiles, ...newFiles]);
-      console.log('INPUT CHANGED', files)
-    }
-  }
+  // const handleChange = function (e) {
```

Comment:
Commented this change handler out as it does not appear to be doing anything

Reply:
This change handler is needed in case a user clicks to upload files rather than drags the files, usually phone users would do that.